### PR TITLE
Pass options from guppy.stream to gulp.src

### LINKED ISSUE #45: Filter subdirectories from guppy.stream.
I have a use case where I want to filter out files from specified subdirectory:
```
gulp.task('pre-commit', function () {
  return guppy.stream('pre-commit')
    .pipe(gulpFilter([ '**/*.js', '!vendor/**/*.js ]))
    .pipe(jshint())
    .pipe(jshint.reporter(stylish))
    .pipe(jshint.reporter('fail'));
});
```
In this situation `gulpFilter` is not working correctly. I did some quick research, and it looks like `guppy.stream()` returns a stream of vinyl files which have their `base` and `relative` fields set to values that are not fully compatible with `gulpFilter`.

```
const gulp = require( 'gulp' );
const guppy = require( 'git-guppy' )( gulp );
const map = require( 'map-stream' );

guppy.stream( 'pre-commit' )
.pipe( map( ( file, cb ) => {
	console.log( file.base, file.relative );

	cb( null, file );
} ) );

/* EXAMPLE OUTPUT
/home/projects/samples/src file1.js
/home/projects/samples/src/lib file2.js
/home/projects/samples/vendor file3.js
*/
```

This example shows that `file.base` is changing, and `file.relative` always points to the file name only. `gulpFilter` uses `file.relative` for multimatching, so no sub-directory checking can be performed.

Could this be changed/fixed, or maybe there is another way to filter files from specified subdirectory?

### PR DESCRIPTION
**Pull request by szymonkups**
Attempt to resolve #45

### CODE REVIEW

File: test/guppy.tests.js [hunk 1]
```diff
@@ -400,6 +401,15 @@ describe('guppy', function () {
       });
     });
 
+    describe('pre-commit gulp options', function () {
+      it('passes options to gulp.src', function () {
+        var options = { base: './' };
+        guppy.stream('pre-commit', options );
+
+        expect(  gulpSrcStub.lastCall.args[1] ).to.equal( options );
```
Comment (therealklanni) [inline test/guppy.tests.js]: Please follow the style :smile: 

No space around parens.

File: lib/get-hook.js [hunk 2]
```diff
@@ -12,8 +12,8 @@ function getIndexed() {
   }).output.split('\n'));
 }
 
-function streamFromIndexed(gulp) {
-  return gulp.src(getIndexed())
+function streamFromIndexed(gulp, options) {
+  return gulp.src(getIndexed(), options )
```
Comment (therealklanni) [inline lib/get-hook.js]: Please remove the extra space here.

File: README.md [hunk 3]
```diff
@@ -156,6 +157,16 @@ gulp.task('pre-commit', function () {
...
+```js
+gulp.task('pre-commit', function () {
+  return guppy.stream('pre-commit', { base: './' })
+    .pipe(gulpFilter(['**/*.js']))
+    .pipe(jshint())
+    .pipe(jshint.reporter(stylish))
+    .pipe(jshint.reporter('fail'));
+});
+```
+
```
Comment (therealklanni) [inline README.md]: I don't think we need an entire example for this, let's just remove it.

File: README.md [hunk 4]
```diff
@@ -134,12 +134,13 @@ gulp.task('pre-push', guppy.src('pre-push', function (files, extra, cb) {
...
 
-#### guppy.stream(*hookName*)
+#### guppy.stream(*hookName, options*)
 
 > Supported hooks: `applypatch-msg`, `commit-msg`, `pre-applypatch`, `pre-commit`, 
 `prepare-commit-msg`
 
-Pass in the name of the git-hook to produce a stream of the related files.
+Pass in the name of the git-hook to produce a stream of the related files. 
+If *options* object is provided, it will be passed down to gulp.src method.
```
Comment (therealklanni) [inline README.md]: Can we reword this to something like "You can pass options as a second argument, please refer to the docs for `gulp.src` for more information on available options." and link to the gulp.src docs?

File: README.md [hunk 5]
```diff
@@ -134,12 +134,13 @@ gulp.task('pre-push', guppy.src('pre-push', function (files, extra, cb) {
 }));
 ```
 
-#### guppy.stream(*hookName*)
+#### guppy.stream(*hookName, options*)
```
Comment (therealklanni) [inline README.md]: This needs to be `guppy.stream(*hookName[, options]*)` to show that options are *optional*.

Comment (therealklanni) [inline README.md]: Let's also update the `guppy.src` line to match.

File: test/guppy.tests.js [hunk 6]
```diff
@@ -400,6 +401,15 @@ describe('guppy', function () {
       });
     });
 
+    describe('pre-commit gulp options', function () {
```
Comment (therealklanni) [inline test/guppy.tests.js]: Can we rename this spec to 'guppy.stream with options'?

File: test/guppy.tests.js [hunk 7]
```diff
@@ -400,6 +401,15 @@ describe('guppy', function () {
...
+      it('passes options to gulp.src', function () {
```
Comment (therealklanni) [inline test/guppy.tests.js]: Can we rename this test to 'should pass options to gulp.src'?

And then can you add another spec for guppy.src that tests the same?

File: test/guppy.tests.js (hunk 1 above)
Comment (therealklanni) [inline test/guppy.tests.js]: Also, I think we should stick to using the sinon-chai syntax.

```js
expect(gulpSrcStub).to.have.been.calledWith(...);
```

**Comment by therealklanni** [comment]
Thanks for the PR, I really appreciate your contribution :+1: 

Just need a few things updated before I can merge.

Have you tried this with your use-case to see if it resolves your absolute path issue?

File: README.md (hunk 5 above)
Comment (szymonkups) [inline README.md]: I don't think it is needed - `guppy.src`. As far as I understand from the docs, `guppy.src` is used to provide file list to `gulp.src`, so library users can pass options directly to `gulp.src`.

File: test/guppy.tests.js (hunk 7 above)
Comment (szymonkups) [inline test/guppy.tests.js]: As in my previous comment - `guppy.src` provides list of files that are passed to `gulp.src`, so options could be provided there without altering this library.

**Comment by szymonkups** [comment]
Thanks for the code review. I've made changes according to Your comments. Please take a look at my responses around `guppy.src` tests and docs if I got this correctly.

After these changes our use-case can be resolved by:

```js
gulp.task('pre-commit', function () {
  return guppy.stream('pre-commit',{base:'./'})
    .pipe(gulpFilter(['**/*.js', '!vendor/**/*.js']))
    .pipe(jshint())
    .pipe(jshint.reporter(stylish))
    .pipe(jshint.reporter('fail'));
});
```

So all the files from `guppy.stream` have base directory set to our current working directory, and relative paths are calculated from that directory. This allows us to use `gulpFilter` to remove files in subdirectories.

File: README.md (hunk 5 above)
Comment (therealklanni) [inline README.md]: Yes, you're correct. Not sure what I was thinking there.

**Comment by therealklanni** [comment]
Looks great @szymonkups. I'll pull this later tonight to just do a quick local test, then I'll get it merged and published.

**Comment by therealklanni** [comment]
Merged by squashing your changes onto master in order to use semantic-release. :metal:

**Comment by therealklanni** [comment]
v1.1.0 should be published now.

**Comment by therealklanni** [comment]
I think the squashing erased your name from the git history, sorry about that. I'm going to add a contributing.md to make sure that future contributors follow the semantic-release git message convention so rewriting commits won't be necessary in the future.

**Comment by szymonkups** [comment]
> I think the squashing erased your name from the git history, sorry about that.

No problem
